### PR TITLE
Update free-courses-si.md

### PR DESCRIPTION
**Pull request by areebniyas**
## What does this PR do?
Add more resources for the Sinhala language.
Add new resource topics and courses.
Fix typos.

### Why is this valuable (or not)?
Help centralize free courses which are not widely available for people that prefer the Sinhala language.

### How do we know it's really free?
Available as YouTube playlists.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by areebniyas** [comment]
@sohansai @eshellman Please merge this pull request.

**Comment by areebniyas** [comment]
@sohansai Could you check and add `ready to merge` label for this PR

**Comment by areebniyas** [comment]
@eshellman Could you review and merge this PR.

File: courses/free-courses-si.md [hunk 1]
```diff
@@ -37,6 +63,7 @@
 * [Introduction to Java](https://www.youtube.com/playlist?list=PLuhSdp06EMkLgaWqSPZKLqePVw-dtqaTT) - Masith Prasanga
 * [Object Oriented Programming ](https://youtube.com/playlist?list=PLqeCu_1ZdDl63h6YR3QsxcGOB7yDS7i3b) - LankaDroid Programming Kuppiya
 * [Sinhala Java Netbeans Lessons](https://youtube.com/playlist?list=PLA3ZeQncjeVu9VHevp2SmPCQ9muVO3fEB) - Chanux Bro
+* [Java Programming Tutorial | Sinhala](https://www.youtube.com/playlist?list=PL495mke12zYANEM9p7JT5-99Yx8Z7z_ib) - CodePRO LK
```
Comment (eshellman) [inline courses/free-courses-si.md:66]:
```suggestion
* [Java Programming Tutorial \| Sinhala](https://www.youtube.com/playlist?list=PL495mke12zYANEM9p7JT5-99Yx8Z7z_ib) - CodePRO LK
```

File: courses/free-courses-si.md [hunk 2]
```diff
@@ -1,30 +1,56 @@
...
-* [Bootstap](https://youtube.com/playlist?list=PLXNgqM9ig24c7IdumyymD9q3e2hsz9U1m&feature=shared) - Udith Sanjaya
+* [Bootstrap](https://youtube.com/playlist?list=PLXNgqM9ig24c7IdumyymD9q3e2hsz9U1m&feature=shared) - Udith Sanjaya
 
 
 ### <a id="csharp"></a>C\#
 
 * [C# Full Course in Sinhala](https://youtube.com/playlist?list=PLvvtf05eMZ2CXD2JdZgSBgyl13ODqHOkO) - Fiqri Ismail
 
 
+### Deep Learning
```
Comment (eshellman) [inline courses/free-courses-si.md:38]: maybe 'artificial intelligence' would be abetter category
It's usually not helpful to add a category for a single resource. Try to find another category, or see if there are other books in our lists that would fit into your category.

File: courses/free-courses-si.md [hunk 3]
```diff
@@ -45,19 +72,27 @@
 * [JavaScript Tutorial in Sinhala](https://youtube.com/playlist?list=PLYmyc7wRFoQjxkHAzHh1UIdU7ZdjTQvQt) - BestJobsLK
 
 
+### Machine Learning
```
Comment (eshellman) [inline courses/free-courses-si.md:75]: maybe "artificial intelligence"?

**Review by eshellman — CHANGES_REQUESTED** [review]

File: courses/free-courses-si.md (hunk 1 above)
Comment (areebniyas) [reply courses/free-courses-si.md]: Resolved with [`01c9770`](https://github.com/EbookFoundation/free-programming-books/pull/10749/commits/01c97709642b1487d34202f5ed6db889df04bc79)

File: courses/free-courses-si.md (hunk 2 above)
Comment (areebniyas) [reply courses/free-courses-si.md]: Resolved with [`01c9770`](https://github.com/EbookFoundation/free-programming-books/pull/10749/commits/01c97709642b1487d34202f5ed6db889df04bc79)

File: courses/free-courses-si.md (hunk 3 above)
Comment (areebniyas) [reply courses/free-courses-si.md]: Resolved with [`01c9770`](https://github.com/EbookFoundation/free-programming-books/pull/10749/commits/01c97709642b1487d34202f5ed6db889df04bc79)

**Comment by areebniyas** [comment]
@eshellman @sohansai Done.

**Review by areebniyas — COMMENTED** [review]
Completed requested changes @eshellman @sohansai

File: courses/free-courses-si.md [hunk 4]
```diff
@@ -10,21 +14,44 @@
...
+
 ### ASP.NET Core
 
 * [WEB API-ASP.NET Core in Sinhala](https://youtube.com/playlist?list=PLvvtf05eMZ2CpeAsq93DqWJHHyvCSa2Qn) - Fiqri Ismail
 
 
 ### Bootstrap
 
-* [Bootstap](https://youtube.com/playlist?list=PLXNgqM9ig24c7IdumyymD9q3e2hsz9U1m&feature=shared) - Udith Sanjaya
+* [Bootstrap](https://youtube.com/playlist?list=PLXNgqM9ig24c7IdumyymD9q3e2hsz9U1m&feature=shared) - Udith Sanjaya
```
Comment (Thenlie) [inline courses/free-courses-si.md:36]:
```suggestion
* [Bootstrap](https://youtube.com/playlist?list=PLXNgqM9ig24c7IdumyymD9q3e2hsz9U1m) - Udith Sanjaya
```

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**